lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -248,6 +248,35 @@ function dosomething_northstar_save_id_field($uid, $northstar_response) {
 }
 
 /**
+ * Save the user's Northstar ID to their local profile.
+ *
+ * @param object $account
+ *   Fully loaded user object.
+ *
+ * @return string
+ *   The Northstar id or empty string on error.
+ */
+function dosomething_northstar_get_id($account = NULL) {
```

Comment:
I tested it — and it fails. That's why I added `try {} } catch (EntityMetadataWrapperException $e) {}` block.